### PR TITLE
chore: make current specs backwards compatible with older versions 

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**

### CODE REVIEW

File: control-plane/agents/src/bin/core/tests/test.rs [hunk 1]
```diff
@@ -4,3 +4,233 @@ mod node;
 mod pool;
 mod volume;
 mod watch;
+
+mod tests_deserializer {
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/tests/test.rs]: maybe create a separate file `deserializer.rs`

File: control-plane/agents/src/bin/core/tests/test.rs [hunk 2]
```diff
@@ -4,3 +4,233 @@ mod node;
...
+                expected: Expected::PoolSpec(PoolSpec {
+                    node: "mayastor-node1".into(),
+                    id: "pool-1-on-mayastor-node1".into(),
+                    disks: vec!["/dev/sdb".into()],
+                    status: PoolSpecStatus::Created(PoolStatus::Online),
+                    labels: {
+                        let mut labels = HashMap::new();
+                        labels.insert(
+                            "openebs.io/created-by".to_string(),
+                            "msp-operator".to_string(),
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/tests/test.rs]: ah, do we need to change this to dsp-operator?

File: control-plane/stor-port/src/types/v0/store/nexus_persistence.rs [hunk 3]
```diff
@@ -7,7 +7,7 @@ use std::fmt::Debug;
 
 /// Definition of the nexus information that gets saved in the persistent
 /// store.
-#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+#[derive(Serialize, Deserialize, Debug, Default, Clone, PartialEq)]
```
Comment (tiagolobocastro) [inline control-plane/stor-port/src/types/v0/store/nexus_persistence.rs]: Eq, PartialEq ?

**Review by tiagolobocastro — COMMENTED** [review]
Does the current code check for dsp-operator or did we keep the msp?

**Comment by Abhinandan-Purkait** [comment]
> Does the current code check for dsp-operator or did we keep the msp?

https://github.com/openebs/mayastor-control-plane/blob/640eba4d0e2031857eea4375c717bb87b66dc7dc/utils/utils-lib/src/constants.rs#L46 It uses `operator-diskpool` now.

File: control-plane/agents/src/bin/core/tests/test.rs (hunk 2 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/tests/test.rs]: We can probably allow this value as well, the key did not change.

File: control-plane/stor-port/src/types/v0/store/nexus_persistence.rs (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/stor-port/src/types/v0/store/nexus_persistence.rs]: Do we need both? Its works without Eq though.

File: control-plane/agents/src/bin/core/tests/test.rs (hunk 2 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/tests/test.rs]: Will raise corresponding PRs to incorporate these legacy changes.

File: control-plane/agents/src/bin/core/tests/test.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/tests/test.rs]: Done

File: control-plane/stor-port/src/types/v0/store/nexus_persistence.rs (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/stor-port/src/types/v0/store/nexus_persistence.rs]: Done.

Comment (tiagolobocastro) [reply control-plane/stor-port/src/types/v0/store/nexus_persistence.rs]: Ah thought there was a linter to check that we have both

File: control-plane/agents/src/bin/core/tests/test.rs (hunk 2 above)
Comment (tiagolobocastro) [reply control-plane/agents/src/bin/core/tests/test.rs]: That sounds good thanks

**Review by tiagolobocastro — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**